components/style/values/computed/length.rs

```diff
@@ -78,7 +78,8 @@ impl ComputeSquaredDistance for CalcLengthOrPercentage {
         // FIXME(nox): This looks incorrect to me, to add a distance between lengths
         // with a distance between percentages.
         Ok(
-            self.unclamped_length().compute_squared_distance(&other.unclamped_length())? +
+            self.unclamped_length().to_f64_px().compute_squared_distance(
+                &other.unclamped_length().to_f64_px())? +
```

Comment:
Actually, this doesn't make any sense. However, while computing the distance or doing interpolation, we only have computed values (i.e. we shouldn't use used values), so this is a temporary? way to define its distance, and the test cases in Gecko use this way to produce the expected values.
We should keep the "FIXME:" because I hope one day we could use a better way to define the distance.